File: backend/src/models/Channel.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface IChannel extends Document {
  name: string;
  serverId: mongoose.Types.ObjectId;
  createdAt: Date;
  updatedAt: Date;
}

const ChannelSchema = new Schema<IChannel>(
  {
    name: {
      type: String,
      required: true,
      trim: true,
    },
    serverId: {
      type: Schema.Types.ObjectId,
      ref: 'Server',
      required: true,
    },
  },
  {
    timestamps: true,
  }
);

export const Channel = mongoose.model<IChannel>('Channel', ChannelSchema);

File: backend/app/models/channel.py
from datetime import datetime
from pydantic import Field
from beanie import Document
from beanie import PydanticObjectId as ObjectId

class Channel(Document):
    name: str
    serverId: ObjectId
    type: str = "text"
    createdAt: datetime = Field(default_factory=datetime.utcnow)
    updatedAt: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "channels"
